Files/ViewModels/ItemViewModel.cs

```diff
@@ -717,15 +713,22 @@ public async void UnloadItemThumbnail(ListedItem item)
                 {
                     item.LoadUnknownTypeGlyph = true;
                 }
-            }, Windows.System.DispatcherQueuePriority.Low);
-            unloadIconSemaphore.Release();
+            };
+            if (CoreApplication.MainView.DispatcherQueue.HasThreadAccess)
```

Comment:
This change forces the icon to be unloaded synchronously. Before I was experiencing issues when switching from Tiles to Grid view mode: this method sometimes completed after the icons for grid view where loaded and the icon got cleared.

Reply:
Looks good